Clarify some subtle behavior for thread gateway events. (#2931)
hammerandchisel/discord-api-docs: docs/resources/Channel.md

## Leave Thread % DELETE /channels/{channel.id#DOCS_RESOURCES_CHANNEL/channel-object}/thread-members/@me

> Removes the current user from a thread. Returns a 204 empty response on success. Also requires the thread is not archived. Fires a [Thread Members Update](#DOCS_TOPICS_GATEWAY/thread-members-update) Gateway event.

Reviewer: ```suggestion
Removes the current user from a thread. The thread must not be archived. Returns a 204 empty response on success. Fires a [Thread Members Update](#DOCS_TOPICS_GATEWAY/thread-members-update) Gateway event.
```
